app/content/profile/index.js

```diff
@@ -125,7 +138,68 @@ function Index() {
   }
 
   const handleImageClick = () => {
-    console.log("TODO: implement and open an image upload modal!");
+    setIsImageUploadModalVisible(true);
+  };
+
+  // Function to handle closing the image upload modal
+  const hideImageUploadModal = () => {
+    setIsImageUploadModalVisible(false);
+  };
+
+  const firebaseProfileImageUpload = async (uri) => {
+    try {
+      setImageUploading(true);
+      // Fetch the image data from the URI
+      const response = await fetch(uri);
+      const blob = await response.blob();
+
+      // Get a reference to the Firebase Storage instance
+      const storage = getStorage();
+
+      // Create a reference to the storage location where the image will be stored
+      const storageRef = ref(storage, userId);
+
+      // Upload the image to Firebase Storage
+      const snapshot = await uploadBytes(storageRef, blob);
+
+      console.log("Uploaded the image blob to the Firebase storage:", snapshot);
+
+      // Get the download URL for the uploaded image
+      const downloadURL = await getDownloadURL(snapshot.ref);
+      await updateDoc(doc(db, "teams", currentTeamId, "users", userId), {
+        pfp: downloadURL,
+      });
+
+      queryClient.invalidateQueries({ queryKey: ["user", { userId }] });
+      setSnackbarMessage("Successfully uploaded the profile picture!");
+      setImageUploading(false);
+
+      return downloadURL;
+    } catch (error) {
+      console.error("Error uploading image to Firebase:", error);
+      setSnackbarMessage("Error uploading profile picture. Please try again.");
+      setImageUploading(false);
+
+      throw error; // Rethrow the error to handle it at the caller's level if needed
+    }
+  };
+
+  // Function to handle uploading image locally
+  const handleImageUpload = async () => {
+    let imageResult = await launchImageLibraryAsync({
+      mediaTypes: MediaTypeOptions.Images,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+
+    // console.log(imageResult);
+
+    if (!imageResult.canceled) {
+      await firebaseProfileImageUpload(imageResult.assets[0].uri);
```

Comment:
I'm not sure how exactly the image upload is triggered, but I noticed that if you upload a new profile photo and then close the modal (without clicking on "update" button), the new profile photo is still saved, although the name is not updated until you click on "update" button. I think that the "update" button should also act as a final confirmation for the new profile photo update, for consistency with the name update.

Screenshots (profile photo updated after closing modal but name did not)
![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/82061589/13584f81-b03f-4cb8-818b-3ebb8143b3b8)
![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/82061589/7dd47ee2-a6bb-4604-bc75-3ffa63fdf9b5)

Reply:
and then have a confirmation box when you close the modal